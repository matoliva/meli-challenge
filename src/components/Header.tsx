import logo from '../assets/logo.png'

interface IProps {
  children?: React.ReactNode
}
export const Header = ({children}: IProps) => {
  return (
    <header className="header" data-testid="header">
      <img className="header__logo" src={logo} alt="logo" />
      {children}
    </header>
  )
}
