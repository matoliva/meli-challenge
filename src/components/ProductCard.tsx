import noImage from '../assets/no-image.png'
import {Link} from 'react-router-dom'
import {IItemMapped} from '../pages/HomePage'

interface IProps {
  item: IItemMapped
}

export const ProductCard = ({item}: IProps) => {
  const {id, title, price, currency, picture, condition, address} = item

  const currencyMap = currency === 'ARS' ? '$' : 'uSd'
  const image = picture ? picture : noImage

  return (
    <Link to={`/item/${id}`} style={{textDecoration: 'none', color: 'black'}}>
      <article
        id={id}
        className="product-card pointer"
        data-testid="product-card"
      >
        <div className="product-card__image">
          <img src={image} alt="product" data-testid="product-card__image" />
        </div>
        <div className="product-card__description">
          <p
            className="product-card__description--price"
            data-testid="product-card__currency"
          >
            {currencyMap} {price}
          </p>
          <p data-testid="product-card__title">{title}</p>
          <p>{condition}</p>
        </div>
        <div className="product-card__location">
          <p data-testid="product-card__location">{address}</p>
        </div>
      </article>
    </Link>
  )
}
