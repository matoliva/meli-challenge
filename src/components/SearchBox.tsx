import logo from '../assets/logo.png'

export const SearchBox = () => {
  return (
    <div className="search-box">
      <img className="search-box__logo" src={logo} alt="logo" />

      <form className="search-box__form">
        <input
          type="text"
          className="input-text"
          placeholder=" Nunca dejes de buscar"
        />
        <div className="search-box__form__icon pointer"></div>
      </form>
    </div>
  )
}
