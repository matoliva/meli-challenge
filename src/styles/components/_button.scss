.btn {
  background-color: #3483fa;
  color: $secondary;
  font-size: medium;
  width: 100%;
  border: none;
  padding: 0.5rem 1rem 0.5rem 1rem;
  border-radius: 4px;

  &:hover {
    background-color: #2968c8;
  }
}
