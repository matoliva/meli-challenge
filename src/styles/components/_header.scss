.header {
  display: flex;
  gap: 1rem;
  width: 100%;
  align-items: center;
  background-color: $primary;
  padding: 0 5rem 0 5rem;
  height: 3rem;

  @media screen and (max-width: $mobile) {
    padding: 0 1rem 0 1rem;
  }

  &__logo {
    max-width: 3rem;
  }
}
