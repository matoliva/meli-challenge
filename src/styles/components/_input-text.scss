.input-text {
  height: 2rem;
  width: 100%;
  border-radius: 2px;
  border: none;
  outline-color: $light-grey;
}
