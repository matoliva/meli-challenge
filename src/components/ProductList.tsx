import {ProductCard} from './ProductCard'
import {IItemMapped} from '../pages/HomePage'

interface IProp {
  items: IItemMapped[]
}

export const ProductList = ({items}: IProp) => {
  return (
    <section data-testid="product-list">
      {items.map(item => (
        <ProductCard key={item.id} item={item} />
      ))}
    </section>
  )
}
