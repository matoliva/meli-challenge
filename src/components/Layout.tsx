import {Header} from './Header'
import {SearchBox} from './SearchBox'

interface IProps {
  children: React.ReactNode
}
export const Layout = ({children}: IProps) => {
  return (
    <div className="main">
      <Header>
        <SearchBox />
      </Header>
      <main className="section">{children}</main>
    </div>
  )
}
