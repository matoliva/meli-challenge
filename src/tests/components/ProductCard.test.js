import {render} from '@testing-library/react'
import {BrowserRouter} from 'react-router-dom'
import {ProductCard} from '../../components/ProductCard'

let mockItem

describe('Product card', () => {
  beforeEach(() => {
    mockItem = {
      id: '1',
      title: 'Zapa Nike',
      currency: 'ARS',
      price: 1000,
      picture: 'http://http2.mlstatic.com/D_807023-MLA48100603525_112021-O.jpg',
      condition: 'Completo unico',
      free_shipping: true,
      address: 'Capital Federal',
    }
  })

  it('renders correctly', () => {
    const {getByTestId} = render(
      <BrowserRouter>
        <ProductCard item={mockItem} />
      </BrowserRouter>,
    )
    const productCard = getByTestId('product-card')
    expect(productCard).toMatchSnapshot()
  })

  it('should show the correct data from props', () => {
    const {getByTestId} = render(
      <BrowserRouter>
        <ProductCard item={mockItem} />
      </BrowserRouter>,
    )
    const productImage = getByTestId('product-card__image')
    const pCurrency = getByTestId('product-card__currency')
    const pTitle = getByTestId('product-card__title')
    const pLocation = getByTestId('product-card__location')

    expect(productImage.src).toContain(mockItem.picture)
    expect(pCurrency.textContent).toBe(`$ ${mockItem.price}`)
    expect(pTitle.textContent).toBe(mockItem.title)
    expect(pLocation.textContent).toBe(mockItem.address)
  })

  it('should show a $ symbol when the currency is ARS', () => {
    const {getByTestId} = render(
      <BrowserRouter>
        <ProductCard item={mockItem} />
      </BrowserRouter>,
    )
    const pCurrency = getByTestId('product-card__currency')

    expect(pCurrency.textContent).toContain('$')
  })

  it('should show a uSd symbol when the currency is Dollar', () => {
    const {getByTestId} = render(
      <BrowserRouter>
        <ProductCard
          item={{
            ...mockItem,
            currency: 'uSd',
            price: 1000,
          }}
        />
      </BrowserRouter>,
    )
    const pCurrency = getByTestId('product-card__currency')

    expect(pCurrency.textContent).toContain('uSd')
  })

  it('should show a default image when there is not image in props', () => {
    const {getByTestId} = render(
      <BrowserRouter>
        <ProductCard
          item={{
            ...mockItem,
            picture: null,
          }}
        />
      </BrowserRouter>,
    )
    const productImage = getByTestId('product-card__image')
    expect(productImage.src).toContain('http://localhost/no-image.png')
  })
})
