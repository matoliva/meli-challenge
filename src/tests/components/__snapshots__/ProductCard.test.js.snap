// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product card renders correctly 1`] = `
<article
  class="product-card pointer"
  data-testid="product-card"
  id="1"
>
  <div
    class="product-card__image"
  >
    <img
      alt="product"
      data-testid="product-card__image"
      src="http://http2.mlstatic.com/D_807023-MLA48100603525_112021-O.jpg"
    />
  </div>
  <div
    class="product-card__description"
  >
    <p
      class="product-card__description--price"
      data-testid="product-card__currency"
    >
      $
       
      1000
    </p>
    <p
      data-testid="product-card__title"
    >
      Zapa Nike
    </p>
    <p>
      Completo unico
    </p>
  </div>
  <div
    class="product-card__location"
  >
    <p
      data-testid="product-card__location"
    >
      Capital Federal
    </p>
  </div>
</article>
`;
